source/javascripts/controllers/_MainController.js.erb

```diff
@@ -338,16 +349,48 @@ import _ from "underscore";
 				}
 			};
 
+			viewModel.openUpdateYmlInRepositoryModal = function() {
+				var variables = appService.getCurrentVariables();
+				appService.validateVariables(variables)
+					.then(function() {
+							viewModel.showUpdateYmlInRepositoryModal = true;
+					})
+			}
+
+			viewModel.closeUpdateYmlInRepositoryModal = function() {
+				$timeout(function() {
+					viewModel.saveProgress.reset();
+					viewModel.showUpdateYmlInRepositoryModal = false;
+				}, 0)
+			}
+
+			viewModel.onSaveToRepoComplete = function() {
+				$timeout(function() {
+					viewModel.closeUpdateYmlInRepositoryModal();
+				}, 2000)
+
+				$timeout(function() {
+					appService.afterAppConfigSaved();
+					viewModel.saveProgress.reset();
+					$rootScope.$emit("MainController::savedFinishedWithSuccess");
+				}, 0)
+			}
+
 			viewModel.save = function(shouldReturnPromise) {
 				if (shouldReturnPromise === undefined) {
 					shouldReturnPromise = false;
 				}
-
+				
 				viewModel.saveProgress.start("<%= data[:strings][:main][:load_progress][:saving] %>");
 
+				if (appService.pipelineConfig.usesRepositoryYml && _.contains(appService.betaTags, 'repository-yml-usage')) {
+					viewModel.openUpdateYmlInRepositoryModal();
+					return;
+				}
+
```

Comment:
This should not be done on all tabs, e. g. not on secrets or stacks tab. Use the switch below.